tests/python_tests/test_rag.py

```diff
@@ -0,0 +1,197 @@
+):
+    if not config:
+        config = TextEmbeddingPipeline.Config()
+
+    pipeline = TextEmbeddingPipeline(models_path, "CPU", config)
+
+    if task == "embed_documents":
+        return pipeline.embed_documents(documents)
+    else:
+        return pipeline.embed_query(documents[0])
+
+
+def run_langchain(
+    models_path: Path,
+    documents: list[str],
+    config: TextEmbeddingPipeline.Config | None = None,
+    task: Literal["embed_documents", "embed_query"] = "embed_documents",
+):
+    if not config:
+        config = TextEmbeddingPipeline.Config()
+
+    encode_kwargs = {
+        "normalize_embeddings": config.normalize,
+        # batch size affects the result
+        "batch_size": len(documents),
+    }
+    if config.pooling_type == TextEmbeddingPipeline.PoolingType.MEAN:
+        encode_kwargs["mean_pooling"] = True
+
+    ov_embeddings = OpenVINOBgeEmbeddings(
+        model_name_or_path=str(models_path),
+        model_kwargs={"device": "CPU"},
+        encode_kwargs=encode_kwargs,
+    )
+
+    # align instructions
+    ov_embeddings.embed_instruction = config.embed_instruction or ""
+    ov_embeddings.query_instruction = config.query_instruction or ""
+
+    if task == "embed_documents":
+        return ov_embeddings.embed_documents(documents)
+    else:
+        return ov_embeddings.embed_query(documents[0])
+
+
+def run_pipeline_with_ref(
+    model_id: str,
+    documents: list[str],
+    config: TextEmbeddingPipeline.Config | None = None,
+    task: Literal["embed_documents", "embed_query"] = "embed_documents",
+):
+    _, _, models_path = download_and_convert_embeddings_models(model_id)
+
+    genai_result = run_genai(models_path, documents, config, task)
+    langchain_result = run_langchain(models_path, documents, config, task)
+
+    np_genai_result = np.array(genai_result)
+    np_langchain_result = np.array(langchain_result)
+
+    max_error = np.abs(np_genai_result - np_langchain_result).max()
+    print(f"Max error: {max_error}")
+    assert (
+        np.abs(np_genai_result - np_langchain_result).max() < 1e-6
+    ), f"Max error: {max_error}"
+
+
+@pytest.mark.precommit
+def test_constructors():
+    model_id = "BAAI/bge-small-en-v1.5"
+    _, _, models_path = download_and_convert_embeddings_models(model_id)
+
+    TextEmbeddingPipeline(models_path, "CPU")
+    TextEmbeddingPipeline(models_path, "CPU", TextEmbeddingPipeline.Config())
+    TextEmbeddingPipeline(
+        models_path,
+        "CPU",
+        TextEmbeddingPipeline.Config(),
+        PERFORMANCE_HINT_NUM_REQUESTS=2,
+    )
+    TextEmbeddingPipeline(
+        models_path,
+        "CPU",
+        normalize=True,
+        pooling_type=TextEmbeddingPipeline.PoolingType.MEAN,
+    )
+    TextEmbeddingPipeline(
+        models_path,
+        "CPU",
+        normalize=True,
+        pooling_type=TextEmbeddingPipeline.PoolingType.MEAN,
+        PERFORMANCE_HINT_NUM_REQUESTS=2,
+    )
```

Comment:
Please parametrize the test with different model configs and model_ids: `TextEmbeddingPipeline(models_path, "CPU", **config)`

Reply:
Test cannot be fully parametrized with `TextEmbeddingPipeline(models_path, "CPU", **config)` as positional args checked as well.